src/Components/WebAssembly/WebAssembly/src/Http/WebAssemblyHttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,169 @@
+        /// <remarks>
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/Request/cache
+        /// </remarks>
+        public static HttpRequestMessage SetBrowserRequestCache(this HttpRequestMessage requestMessage, BrowserRequestCache requestCache)
+        {
+            if (requestMessage is null)
+            {
+                throw new ArgumentNullException(nameof(requestMessage));
+            }
+
+            var stringOption = requestCache switch
+            {
+                BrowserRequestCache.Default => "default",
+                BrowserRequestCache.NoStore => "no-store",
+                BrowserRequestCache.Reload => "reload",
+                BrowserRequestCache.NoCache => "no-cache",
+                BrowserRequestCache.ForceCache => "force-cache",
+                BrowserRequestCache.OnlyIfCached => "only-if-cached",
+                _ => throw new InvalidOperationException($"Unsupported enum value {requestCache}.")
+            };
+
+            return SetBrowserFetchOption(requestMessage, "cache", stringOption);
+        }
+
+        /// <summary>
+        /// Configures a value for the 'mode' option for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="requestMode">The <see cref="BrowserRequestMode"/>.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>\
+        /// <remarks>
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/Request/mode
+        /// </remarks>
+        public static HttpRequestMessage SetBrowserRequestMode(this HttpRequestMessage requestMessage, BrowserRequestMode requestMode)
+        {
+            if (requestMessage is null)
+            {
+                throw new ArgumentNullException(nameof(requestMessage));
+            }
+
+            var stringOption = requestMode switch
+            {
+                BrowserRequestMode.SameOrigin => "same-origin",
+                BrowserRequestMode.NoCors => "no-cors",
+                BrowserRequestMode.Cors => "cors",
+                BrowserRequestMode.Navigate => "navigate",
+                _ => throw new InvalidOperationException($"Unsupported enum value {requestMode}.")
+            };
+
+            return SetBrowserFetchOption(requestMessage, "mode", stringOption);
+        }
+
+        /// <summary>
+        /// Configures a value for the 'integrity' option for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="integrity">The subresource integrity descriptor.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>
+        /// <remarks>
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/Request/integrity
+        /// </remarks>
+        public static HttpRequestMessage SetBrowserIntegrity(this HttpRequestMessage requestMessage, string integrity)
```

Comment:
```suggestion
        public static HttpRequestMessage SetBrowserRequestIntegrity(this HttpRequestMessage requestMessage, string integrity)
```

All the preceding methods have "Request" in the name

Reply:
Also helps to clarify it's not the integrity of the browser we're checking, but of this individual request